### PR TITLE
sink(ticdc): add transaction-atomicity parameter to SinkURI

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref #5231

### What is changed and how it works?
1. Add `transaction-atomicity` parameter to SinkURI
2. Use `validateAndAdjust` method to parse and verify sinkURI's parameters uniformly.
3. Add splitTxn field to flowController to accommodate different transaction atomicity.
4. Add defensive checks in sinkNode.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

##### Will it cause performance regression or break compatibility?
No

##### Do you need to update user documentation, design documentation or monitoring documentation?
Yes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: pkg/config/sink.go [hunk 1]
```diff
@@ -15,16 +15,48 @@ package config
...
+
+	// noneTxnAtomicity means atomicity of transactions is not guaranteed
+	noneTxnAtomicity AtomicityLevel = "none"
+
+	// tableTxnAtomicity means atomicity of single table transactions is guaranteed.
+	tableTxnAtomicity AtomicityLevel = "table"
+
+	// globalTxnAtomicity means atomicity of cross table transactions is guaranteed, which
+	// is currently not supported by TiCDC.
+	globalTxnAtomicity AtomicityLevel = "global"
```
Comment (overvenus) [inline pkg/config/sink.go:45]: Prefer remove or comment out if it is not supported.

```suggestion
	// globalTxnAtomicity AtomicityLevel = "global"
```

Comment (CharlesCheung96) [reply pkg/config/sink.go:45]: Good suggestion!

File: cdc/sink/mq/mq.go [hunk 2]
```diff
@@ -386,10 +386,6 @@ func NewKafkaSaramaSink(ctx context.Context, sinkURI *url.URL,
 		return nil, cerror.WrapError(cerror.ErrKafkaInvalidConfig, err)
 	}
 
-	if err := replicaConfig.ApplyProtocol(sinkURI).Validate(); err != nil {
```
Comment (nongfushanquan) [inline cdc/sink/mq/mq.go:389]: why did you remove the validation?

Comment (CharlesCheung96) [reply cdc/sink/mq/mq.go:389]: Move this to: https://github.com/pingcap/tiflow/pull/6038/files#diff-1aa9ba3321cb0edb368a32b5084da34322b0f9d6a600019beb724ab63bc5e071R162

**Review by nongfushanquan — APPROVED** [review]

File: pkg/config/sink.go [hunk 3]
```diff
@@ -15,16 +15,48 @@ package config
...
 
-// DefaultMaxMessageBytes sets the default value for max-message-bytes
+// DefaultMaxMessageBytes sets the default value for max-message-bytes.
 const DefaultMaxMessageBytes = 10 * 1024 * 1024 // 10M
 
+// AtomicityLevel represents the atomicity level of a changefeed.
+type AtomicityLevel string
+
+const (
+	// unknowTxnAtomicity is the default atomicity level, which is invalid and will
```
Comment (maxshuang) [inline pkg/config/sink.go:33]: May be we can define a default variable like 'defaultAtomicityLevel' to make the system behavior more clear.

Comment (CharlesCheung96) [reply pkg/config/sink.go:33]: Done, PTAL!

File: cdc/processor/pipeline/sink.go [hunk 4]
```diff
@@ -360,3 +362,23 @@ func (n *sinkNode) releaseResource(ctx context.Context) error {
...
 }
+
+func (n *sinkNode) checkSplitTxn(e *model.PolymorphicEvent) {
+	ta := n.replicaConfig.Sink.TxnAtomicity
+	ta.Validate()
+	if ta.ShouldSplitTxn() {
+		return
+	}
+
+	// Check that BatchResolved events and RowChangedEvent events with `SplitTxn==true`
```
Comment (asddongmen) [inline cdc/processor/pipeline/sink.go:373]: please add comments to tell that this situation should never happen normally.

File: cdc/processor/pipeline/sink.go [hunk 5]
```diff
@@ -360,3 +362,23 @@ func (n *sinkNode) releaseResource(ctx context.Context) error {
 	n.flowController.Abort()
 	return n.sink.Close(ctx)
 }
+
+func (n *sinkNode) checkSplitTxn(e *model.PolymorphicEvent) {
```
Comment (asddongmen) [inline cdc/processor/pipeline/sink.go:366]: please add comments to tell the purpose of this method.

File: cdc/sink/flowcontrol/flow_control.go [hunk 6]
```diff
@@ -92,8 +100,14 @@ func (c *TableFlowController) Consume(
 ) error {
 	commitTs := msg.CRTs
 	lastCommitTs := atomic.LoadUint64(&c.lastCommitTs)
-	blockingCallBack := func() error {
-		err := callBack(c.batchID)
+	blockingCallBack := func() (err error) {
+		if commitTs != lastCommitTs || c.splitTxn {
```
Comment (asddongmen) [inline cdc/sink/flowcontrol/flow_control.go:104]: Can we just use `commitTs > lastCommitTs` ? It would be more clear.

**Review by asddongmen — APPROVED** [review]
